stream.go

```diff
@@ -217,6 +218,7 @@ func (st *Stream) produceKVs(ctx context.Context, threadId int) error {
 		for itr.Seek(kr.left); itr.Valid(); {
 			// it.Valid would only return true for keys with the provided Prefix in iterOpts.
 			item := itr.Item()
+			atomic.AddUint64(&st.scanned, uint64(len(item.key)+len(item.vptr)+len(item.val)))
```

Comment:
vptr could be 8 bytes while the actual value size could be 1 mb. You might want to decode the value pointer and fetch the correct length.

Reply:
We are comparing `st.scanned` with `uncompressedSize` that stores the size of SST (i.e it considers only the vptr for bigger values). Hence, we should not decode the `vptr` to get the actual length.